src/firstboot/firstboot.c

```diff
@@ -308,28 +308,36 @@ static int prompt_locale(void) {
         return 0;
 }
 
-static int process_locale(void) {
-        const char *etc_localeconf;
+static int process_locale(int rfd) {
+        _cleanup_close_ int pfd = -EBADF;
         char* locales[3];
         unsigned i = 0;
         int r;
 
-        etc_localeconf = prefix_roota(arg_root, "/etc/locale.conf");
-        if (laccess(etc_localeconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming locale information has been configured.",
-                          etc_localeconf);
+        r = chase_symlinks_at(rfd, "etc", CHASE_AT_RESOLVE_IN_ROOT|CHASE_NONEXISTENT, NULL, &pfd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to resolve %s/etc/: %m", FORMAT_FD_PATH(rfd));
```

Comment:
I wonder if we should have CHASE_MKDIR flag btw, which whenever we hit a dir in the path will create it as dir. We could set that here and then be compatible with cases where /etc/ is missing initially.

Reply:
That would mean we'd change behavior and would always create /etc/, whereas now we only create /etc/ when we're going to write to a file in it.

Reply:
that doesn't sound too bad to me... /etc/ is really something that should exist if it is missing...

Reply:
Still not sure how this should work exactly though. Let's say we have a path with ".." in it. If we use CHASE_MKDIR, we'll end up creating unnecessary directories. I'm pretty sure we should first fully resolve the path and only after doing that create any missing directories.